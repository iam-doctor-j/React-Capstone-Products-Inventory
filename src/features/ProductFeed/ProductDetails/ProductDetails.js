import React from 'react';
import Card from "react-bootstrap/Card";
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { fetchProductsFromDb, viewProductFromDb } from '../ProductFeedSlice';

class ProductDetails extends React.Component {
    
    componentDidMount() {
        this.props.dispatch(viewProductFromDb(this.props.match.params.id));
        if(!this.props.product) {
            this.props.dispatch(fetchProductsFromDb());
        }
    }

    render() {
        return(
            <div className="container">
                <div className="row justify-content-center">
                    <Card className="col-12 col-md-8 col-lg-6">
                        <Card.Body>
                            <Card.Title className="mb-4">
                                <div className="d-flex">
                                    <span className="heading-text">Details</span>
                                    <a className="ml-auto" onClick={this.props.history.goBack} style={{cursor: 'pointer', color: 'var(--secondary)'}}><i className="fas fa-arrow-circle-left"></i></a>
                                </div>
                            </Card.Title>
                            <div className="mb-3">
                                <div className="label-text">Product Name</div>
                                <div>{this.props.product && this.props.product.name.value}</div>
                            </div>
                            <div className="mb-3">
                                <div className="label-text">Product Description</div>
                                <div>{this.props.product && this.props.product.description.value}</div>
                            </div>
                            <div className="mb-3">
                                <div className="label-text">Product Manufacturer</div>
                                <div>{this.props.product && this.props.product.manufacturer.value}</div>
                            </div>
                            <div className="mb-3">
                                <div className="label-text">Product Price</div>
                                <div>{this.props.product && this.props.product.price.value}</div>
                            </div>
                            <div className="mb-3">
                                <div className="label-text">Product Quantity</div>
                                <div>{this.props.product && this.props.product.quantity.value}</div>
                            </div>
                        </Card.Body>
                    </Card>
                </div>
            </div>
        );
    }
}

export default connect((state, ownProps) => {
    console.log(ownProps);
    console.log(state);
    let id = ownProps.match.params.id;
    let product;
    if(state.productFeed.products && state.productFeed.products.length!==0) {
        product = _.find(state.productFeed.products, product => product.id === id);
    }
    return {
        product
    }
})(ProductDetails);