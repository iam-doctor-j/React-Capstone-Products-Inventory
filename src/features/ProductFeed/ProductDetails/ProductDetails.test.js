import React from 'react';
import { shallow } from 'enzyme';
import { ProductDetails } from './ProductDetails';
import Card from 'react-bootstrap/Card';

describe('Test ProductDetails with shallow rendering', () => {
    let wrapper;

    const strProd = `{
        "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
        "views": 6,
        "name": {
          "value": "Laptop",
          "visible": true
        },
        "description": {
          "value": "A simple Laptop",
          "visible": true
        },
        "manufacturer": {
          "value": "Lenovo",
          "visible": true
        },
        "price": {
          "value": 64999,
          "visible": true
        },
        "quantity": {
          "value": 1,
          "visible": true
        }
      }`

    const props = {
        product: JSON.parse(strProd),
        dispatch: () => {},
        history: () => {},
        match: {
            params: {
                id: "63796e43-f23b-542e-99a7-11db520c7bb6",
            }
        }
    }

    beforeEach(() => {
        wrapper = shallow(<ProductDetails {...props}/>);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render one div with class container', () => {
        expect(wrapper.find('div.container').length).toEqual(1);
    });

    it('should render one card', () => {
        expect(wrapper.find(Card).length).toEqual(1);
    });

    afterEach(() => {
        wrapper.unmount();
    })
});