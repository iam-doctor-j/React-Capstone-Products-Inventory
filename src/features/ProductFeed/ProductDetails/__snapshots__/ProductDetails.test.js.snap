// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductDetails with shallow rendering should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center"
  >
    <Card
      body={false}
      className="col-12 col-md-8 col-lg-6"
    >
      <CardBody>
        <CardTitle
          className="mb-4"
        >
          <div
            className="d-flex"
          >
            <span
              className="heading-text"
            >
              Details
            </span>
            <a
              className="ml-auto"
              style={
                Object {
                  "color": "var(--secondary)",
                  "cursor": "pointer",
                }
              }
            >
              <i
                className="fas fa-arrow-circle-left"
              />
            </a>
          </div>
        </CardTitle>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Product Name
          </div>
          <div>
            Laptop
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Product Description
          </div>
          <div>
            A simple Laptop
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Product Manufacturer
          </div>
          <div>
            Lenovo
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Product Price
          </div>
          <div>
            64999
          </div>
        </div>
        <div
          className="mb-3"
        >
          <div
            className="label-text"
          >
            Product Quantity
          </div>
          <div>
            1
          </div>
        </div>
      </CardBody>
    </Card>
  </div>
</div>
`;
